linkatos.py

```diff
@@ -5,75 +5,106 @@
     """
-        Receives commands directed at the bot and determines if they
-        are valid commands. If so, then acts on the commands. If not,
-        returns back what it needs for clarification.
+        Receives a link.
+        Stores it in the db.
+        Sends a confirmation message.
     """
-    response = "Not sure what you mean. Use the *" + EXAMPLE_COMMAND + \
-               "* command with numbers, delimited by spaces."
-    if command.startswith(EXAMPLE_COMMAND) :
-        response = "Sure...write some more code then I can do that!"
-    
-    bot_message(channel, response)
+
+    error = db.store(link)
+    message = "Success! The link has been stored"
+
+    if error :
+        message = "ERROR: the link could not be stored"
+
+    bot_says(channel, message)
+
+    return None
+
+
+def message_contains_link (message) :
+    """
+    Returns a link if it matches the regex
+    """
+    website_pattern = "https?://\S+(\s|$)"
+    prog = re.compile(website_pattern)
+    return prog.search(output['text']).strip() 
 
 
 def parse_slack_output(slack_rtm_output):
     """
         The Slack Real Time Messaging API is an events firehose.
-        this parsing function returns None unless a message is
-        directed at the Bot, based on its ID.
+        this parsing function returns None unless
+        someone posts a website link starting with httpS?//
+
+        In these situations the function will return the link and the channel 
+        where the message was posted.
+
     """
+
+    # default outcome
+    link = None
+    channel = None
+
     output_list = slack_rtm_output
-    print output_list
+    print output_list  # print the list of outputs to get them on screen
+
     if output_list and len(output_list) > 0 :
         for output in output_list :
-            if output and 'text' in output and AT_BOT in output['text'] :
-                slack_client.api_call("chat.postMessage", 
-                    channel = output['channel'], 
-                    text = "r u talkin' to me?",
-                    as_user = True)
-                return output['text'].split(AT_BOT)[1].strip().lower(), \
-                     output['channel']
-            elif output and \
-                    'text' in output and \
-                    prog.search(output['text']) != None and \
-                    output['user'] !=  BOT_ID :
-                response = "It looks like you posted a link. /n \
-                    The link is: " + prog.search(output['text']).group(0)
-                bot_message(output['channel'], response)
-                # return text after the @ mention, whitespace removed
-                return output['text'].strip().lower(), output['channel']
-    return None, None
-
-
-if __name__ == "__main__" :
+            # when there is a message then get the channel
+            if channel : channel = output['channel']
+
+            if output and output['text'] and output['user'] != BOT_ID :
```

Comment:
`output` diria que pot ser `None`, i aleshores no entraria a l'`if`, però no ho tinc clar, això ve del codi original i tot i que em va fer pensar una estona si era necessari de moment ho vaig deixar tal qual.

Provaré de treure-ho.

hm no sabia que existia `is not` m'agrada més sí.